File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/02_Conditional_Statements_Exercise/08_Lunch_Break.js
function lunchBreak(input) {
    let showName = input[0];  // "Game of Thrones"
    let showDuration = Number(input[1]);  // 60
    let breakDuration = Number(input[2]);  // 96

    let lunchDuration = breakDuration / 8;
    let restingDuration = breakDuration / 4;

    let usedTime = lunchDuration + restingDuration;
    // breakDuration -= lunchDuration;
    // breakDuration -= restingDuration;

    if ((breakDuration - usedTime) >= showDuration) {
        console.log(`You have enough time to watch ${showName} and left with ${Math.ceil(breakDuration - showDuration)} minutes free time.`);
    } else {
        console.log(`You don't have enough time to watch ${showName}, you need ${Math.ceil(showDuration - breakDuration)} more minutes.`);
    }


}

// lunchBreak(["Game of Thrones", "60", "96"]);
// lunchBreak(["Teen Wolf", "48", "60"]);

File: Python_Courses/Python_Basics/Preparation/02_Conditional_Statements_Exercise/08_Lunch_Break.py
import math

LUNCH_TIME_AS_MULTIPLIER = 1 / 8
REST_TIME_AS_MULTIPLIER = 1 / 4

movie_series = input()
episode_length = int(input())
break_length = int(input())

break_extra_length = (1 - (LUNCH_TIME_AS_MULTIPLIER + REST_TIME_AS_MULTIPLIER)) * break_length

if break_extra_length >= episode_length:
    print(f"You have enough time to watch {movie_series} "
          f"and left with {math.ceil(break_extra_length - episode_length)} minutes free time.")
else:
    print(f"You don't have enough time to watch {movie_series}, "
          f"you need {math.ceil(episode_length - break_extra_length)} more minutes.")
